koku/masu/database/trino_sql/reporting_awscostentrylineitem_summary_by_ec2_compute_p.sql

```diff
@@ -79,82 +101,74 @@ SELECT uuid() as uuid,
+               THEN 0.0
+               ELSE lineitem_usageamount
+           END
+       ) as usage_amount,
+       max(lineitem_normalizationfactor) as normalization_factor,
+       sum(lineitem_normalizedusageamount) as normalized_usage_amount,
+       max(lineitem_currencycode) as currency_code,
+       max(lineitem_unblendedrate) as unblended_rate,
+       /* SavingsPlanCoveredUsage entries have corresponding SavingsPlanNegation line items
            that offset that cost.
            https://docs.aws.amazon.com/cur/latest/userguide/cur-sp.html
-        */
-        sum(
-            CASE
-                WHEN lineitem_lineitemtype='SavingsPlanCoveredUsage'
-                THEN 0.0
-                ELSE lineitem_unblendedcost
-            END
-        ) as unblended_cost,
-        max(lineitem_blendedrate) as blended_rate,
+       */
+       sum(
+           CASE
+               WHEN lineitem_lineitemtype='SavingsPlanCoveredUsage'
+               THEN 0.0
+               ELSE lineitem_unblendedcost
+           END
+       ) as unblended_cost,
+       max(lineitem_blendedrate) as blended_rate,
         /* SavingsPlanCoveredUsage entries have corresponding SavingsPlanNegation line items
            that offset that cost.
            https://docs.aws.amazon.com/cur/latest/userguide/cur-sp.html
         */
-        sum(
-            CASE
-                WHEN lineitem_lineitemtype='SavingsPlanCoveredUsage'
-                THEN 0.0
-                ELSE lineitem_blendedcost
-            END
-        ) as blended_cost,
-        sum(savingsplan_savingsplaneffectivecost) as savingsplan_effective_cost,
-        sum(
-            CASE
-                WHEN lineitem_lineitemtype='Tax'
-                OR   lineitem_lineitemtype='Usage'
-                THEN lineitem_unblendedcost
-                ELSE savingsplan_savingsplaneffectivecost
-            END
-        ) as calculated_amortized_cost,
-        sum(pricing_publicondemandcost) as public_on_demand_cost,
-        max(pricing_publicondemandrate) as public_on_demand_rate
-
-    FROM hive.{{schema | sqlsafe}}.aws_line_items_daily
+       sum(
+           CASE
+               WHEN lineitem_lineitemtype='SavingsPlanCoveredUsage'
+               THEN 0.0
+               ELSE lineitem_blendedcost
+           END
+       ) as blended_cost,
+       sum(savingsplan_savingsplaneffectivecost) as savingsplan_effective_cost,
+       sum(
+           CASE
+               WHEN lineitem_lineitemtype='Tax'
+               OR lineitem_lineitemtype='Usage'
+               THEN lineitem_unblendedcost
+               ELSE savingsplan_savingsplaneffectivecost
+           END
+       ) as calculated_amortized_cost,
+       sum(pricing_publicondemandcost) as public_on_demand_cost,
+       max(pricing_publicondemandrate) as public_on_demand_rate
+    FROM hive.{{schema | sqlsafe}}.aws_line_items_daily as lid
+        JOIN cte_latest_values AS cte_l
+        ON lid.lineitem_resourceid = cte_l.resource_id
```

Comment:
Suggestion: 

I think this join should probably move up a level, maybe something similar to this;

```
CROSS JOIN cte_pg_enabled_keys AS pek
JOIN cte_latest_values AS cte_l ON ds.resource_id = cte_l.resource_id
LEFT JOIN postgres.{{schema | sqlsafe}}.reporting_awsaccountalias AS aa
```


and then use `cte_latest_values` results in the outer select statement

```
SELECT uuid() as uuid,
    -- ...
    resource_id,
    cte_l.instance_name,
    cte_l.instance_type,
    operating_system,
    region,
    vcpu,
    memory,
    cast(
        map_filter(
            cast(json_parse(cte_l.tags) as map(varchar, varchar)),
            (k,v) -> contains(pek.keys, k)
        ) as json
    ) as tags,
    cast(json_parse(cte_l.cost_category) as json) as cost_category,
```